lib/surface/view.ex

```diff
@@ -60,21 +60,36 @@ defmodule Surface.View do
   defmacro __before_compile__(env) do
     root = Module.get_attribute(env.module, :surface_view_root)
 
-    for {name, path} <- templates(env.module, root) do
-      ast =
-        path
-        |> File.read!()
-        |> Surface.Compiler.compile(1, env, path)
-        |> Surface.Compiler.to_live_struct()
+    render_funs =
+      for {name, path} <- templates(env.module, root) do
+        ast =
+          path
+          |> File.read!()
+          |> Surface.Compiler.compile(1, env, path)
+          |> Surface.Compiler.to_live_struct()
 
+        quote do
+          @file unquote(path)
+          @external_resource unquote(path)
+          defp __render_surface__(unquote(name), var!(assigns)) do
+            unquote(ast)
+          end
+        end
+      end
+
+    recompilation_helper =
       quote do
-        @file unquote(path)
-        @external_resource unquote(path)
-        defp render_surface(unquote(name), var!(assigns)) do
-          unquote(ast)
+        defmodule SurfaceRecompilationHelper do
```

Comment:
@guisehn have you tried to add `defoverridable` before defining the function, like:

```elixir
defoverridable __mix_recompile__?: 0

def __mix_recompile__? do
  ...
end
```

Reply:
Nope. I'll try that. :)

Reply:
@msaraiva adjusted on https://github.com/surface-ui/surface/pull/548/commits/101b24a937e24e632793cd52ec5417f33939ddc3

It worked, we just need to make sure that `use Surface.View` is called *after* `use Phoenix.View` on the templates, because `Phoenix.View` doesn't do `defoverridable` when defining `__mix_recompile__?`.